tests/components/litterrobot/test_vacuum.py

```diff
@@ -0,0 +1,92 @@
+"""Test the Litter-Robot vacuum entity."""
+from datetime import timedelta
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components import litterrobot
+from homeassistant.components.litterrobot.hub import REFRESH_WAIT_TIME
+from homeassistant.components.vacuum import (
+    ATTR_PARAMS,
+    DOMAIN as PLATFORM_DOMAIN,
+    SERVICE_SEND_COMMAND,
+    SERVICE_START,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_DOCKED,
+)
+from homeassistant.const import ATTR_COMMAND, ATTR_ENTITY_ID
+from homeassistant.util.dt import utcnow
+
+from .common import CONFIG
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+ENTITY_ID = "vacuum.test_litter_box"
+
+
+async def setup_hub(hass, mock_hub):
+    """Load the Litter-Robot vacuum platform with the provided hub."""
+    hass.config.components.add(litterrobot.DOMAIN)
+    entry = MockConfigEntry(
+        domain=litterrobot.DOMAIN,
+        data=CONFIG[litterrobot.DOMAIN],
+    )
+
+    with patch.dict(hass.data, {litterrobot.DOMAIN: {entry.entry_id: mock_hub}}):
```

Comment:
We don't want to interact with integration details in the tests.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations